driver-workload-executor/src/main/com/mongodb/workload/WorkloadExecutor.java

```diff
@@ -0,0 +1,138 @@
+ *  org.mongodb.workload.spec: a JSON representation of the the workload
+ *  org.mongodb.workload.output.directory: the output directory in which to write the results.json
+ */
+public class WorkloadExecutor {
+    private static volatile boolean done;
+    private static final AtomicInteger numSuccesses = new AtomicInteger();
+    private static final AtomicInteger numFailures = new AtomicInteger();
+    private static final AtomicInteger numErrors = new AtomicInteger();
+
+    /**
+     * The main entry point
+     *
+     * @param args no arguments are expected
+     */
+    public static void main(String[] args) {
+        String outputDirectory = System.getProperty("org.mongodb.workload.output.directory") == null
+                ? System.getProperty("user.dir") : System.getProperty("org.mongodb.workload.output.directory");
+        
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                done = true;
+                writeResults(outputDirectory);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }));
+
+        String connectionString = System.getProperty("org.mongodb.test.uri");
+        if (connectionString == null) {
+            connectionString = "mongodb://localhost";
+        }
+
+        MongoClient mongoClient = MongoClients.create(connectionString);
+
+        String workloadSpec = System.getProperty("org.mongodb.workload.spec");
+        BsonDocument workload = BsonDocument.parse(workloadSpec);
+
+        MongoDatabase database = mongoClient.getDatabase(workload.getString("database").getValue());
+        MongoCollection<BsonDocument> collection = database.getCollection(workload.getString("collection").getValue(),
+                BsonDocument.class);
+
+        if (workload.containsKey("testData")) {
+            List<BsonDocument> documents = new ArrayList<>();
+            for (BsonValue document : workload.getArray("testData")) {
+                documents.add(document.asDocument());
+            }
+
+            collection.withWriteConcern(WriteConcern.MAJORITY).drop();
+            collection.withWriteConcern(WriteConcern.MAJORITY).insertMany(documents);
+        }
+
+        JsonPoweredCrudTestHelper helper = new JsonPoweredCrudTestHelper("Workload executor", database, collection);
+
+        BsonArray operations = workload.getArray("operations");
+
+        outer:
+        while (!done) {
+            for (BsonValue cur : operations.getValues()) {
+                try {
+                    if (done) {
+                        break outer;
+                    }
+                    BsonDocument operation = cur.asDocument().clone();
+                    BsonValue expectedResult = operation.get("result");
+                    BsonDocument resultDocument = helper.getOperationResults(operation);
+
+                    if (expectedResult != null && !resultDocument.get("result").equals(expectedResult)) {
```

Comment:
Can expectedResult be null and the operation return a result?

Reply:
Yes, I think that's allowed. It means that the runner shouldn't check the result and just treat any normal return as a success.